scio-core/src/main/scala/com/spotify/scio/io/Tap.scala

```diff
@@ -77,11 +79,12 @@ case class BigQueryTap(table: TableReference) extends Tap[TableRow] {
 }
 
 /** Tap for object files on local file system or GCS. */
-case class ObjectFileTap[T: ClassTag](path: String) extends Tap[T] {
+case class ObjectFileTap[T: ClassTag](path: String, coder: Coder[T] = null) extends Tap[T] {
```

Comment:
In `SCollection` and `ScioContext` we have `CoderRegistry` available - `Tap`s are totally separated - thus user might want to decide, also I later use it to to pass coder from `saveAsObject` to point to the same coder without the need to create another `CoderRegistry` instance just to find something that is already available.